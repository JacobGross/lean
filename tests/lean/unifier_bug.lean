import logic

theorem test {A B : Type} {a : A} {b : B} (H : a == b) :
           eq.rec_on (heq.type_eq H) a = b
:=
-- Remark the error message should not occur in the token theorem
heq.rec_on H rfl
