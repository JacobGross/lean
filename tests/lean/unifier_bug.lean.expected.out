unifier_bug.lean:7:0: error: type mismatch at application
  heq.rec_on H rfl
term
  rfl
has type
  eq.rec_on (heq.type_eq H) a = eq.rec_on (heq.type_eq H) a
but is expected to have type
  eq.rec_on (heq.type_eq H) a = b
